src/app/main.rs

```diff
@@ -514,11 +520,22 @@ where
 
             // Routes requests to their original destination endpoints. Used as
             // a fallback when service discovery has no endpoints for a destination.
+            //
+            // If the `l5d-require-id` header is present, then that identity is
+            // used as the server name when connecting to the endpoint.
             let orig_dst_router = svc::builder()
                 .layer(router::layer(
                     router::Config::new("out ep", capacity, max_idle_age),
                     |req: &http::Request<_>| {
-                        let ep = outbound::Endpoint::from_orig_dst(req);
+                        let ep = outbound::Endpoint::from_orig_dst(req).and_then(|mut ep| {
+                            if let Some(require_id) =
+                                identity_from_header(req, super::L5D_REQUIRE_ID)
+                            {
+                                debug!("outbound ep require id={:?}", require_id);
+                                ep.identity = Conditional::Some(require_id);
```

Comment:
How do you feel about moving this logic (and the `identity_from_header` utility) into the `outbound` module? This accomplishes a few things:

1. `main` is less verbose (there's already a lot in here_;
2. this logic only ever applies on the outbound side, so it need not be generalized;
3. identity is only constructed from a header in this context, so it need not be generalized

Reply:
Something like this? I'd like to still show we're doing an additional check for the presence of a header, but agree it's overly verbose right now.

```
|req: &http::Request<_>| {
    let ep = outbound::Endpoint::from_orig_dst(req).and_then(|mut ep| {
        let ep = outbound::should_require_identity(ep, req);
        Some(ep)
    });
    debug!("outbound ep={:?}", ep);
    ep
},
```

Reply:
Well something more like:

```
|req: &http::Request<_>| {
    let ep = outbound::Endpoint::from_orig_dst(req).map(|ep| {
        outbound::should_require_identity(ep, req)
    });
    debug!("outbound ep={:?}", ep);
    ep
},
```